NGitLab/Impl/SearchClient.cs

```diff
@@ -0,0 +1,77 @@
+﻿using System;
+using NGitLab.Models;
+
+namespace NGitLab.Impl
+{
+    public class SearchClient : ISearchClient
+    {
+        private readonly API _api;
+        private readonly string _baseUrl;
+
+        public SearchClient(API api, string baseUrl)
+        {
+            _api = api;
+            _baseUrl = baseUrl;
+        }
+
+        public GitLabCollectionResponse<SearchBlob> GetBlobs(SearchQuery query)
+        {
+            var url = CreateGetUrl("blobs", query);
+            return _api.Get().GetAllAsync<SearchBlob>(url);
+        }
+
+        private string CreateGetUrl(string scope, SearchQuery query)
+        {
+            var url = _baseUrl;
+
+            url = Utils.AddParameter(url, "scope", scope);
+            url = Utils.AddParameter(url, "search", query.Search);
+
+            if (query.PerPage.HasValue)
+            {
+                url = Utils.AddParameter(url, "per_page", query.PerPage);
+            }
+
+            if (query.Confidential.HasValue)
+            {
+                if (!string.Equals(scope, "issues", StringComparison.Ordinal))
```

Comment:
Can't we let GitLab do the check?

Reply:
Yes, it will just ignore if not of the right type